lib/image.js

```diff
@@ -1,12 +1,71 @@
-"use strict";
-var Promise = require('bluebird');
+'use strict';
+var Cesium = require('cesium');
 var fs = require('fs-extra');
 var path = require('path');
+var PNG = require('pngjs').PNG;
+var Promise = require('bluebird');
 
 var fsReadFile = Promise.promisify(fs.readFile);
 
+var WebGLConstants = Cesium.WebGLConstants;
+
 module.exports = loadImage;
 
+/**
+ * Load an image file and get information about it.
+ *
+ * @param {String} imagePath Path to the image file.
+ * @returns {Promise} A promise resolving to the image information, or undefined if the file doesn't exist.
+ *
+ * @private
+ */
+function loadImage(imagePath) {
+    return fsReadFile(imagePath)
+        .then(function(data) {
+            var extension = path.extname(imagePath);
+            var uriType = getUriType(extension);
+            var uri = uriType + ';base64,' + data.toString('base64');
```

Comment:
Should this include a check on the buffer length for large textures? Otherwise, maybe we should add a troubleshooting section in the README.md with things to try, like writing out separate images if there's a `buffer.toString` error.